broker/src/test/java/io/zeebe/broker/system/configuration/BrokerCfgTest.java

```diff
@@ -480,6 +480,42 @@ public void shouldSetBackpressureConfig() {
     assertThat(backpressure.getAlgorithm()).isEqualTo(LimitAlgorithm.GRADIENT);
   }
 
+  @Test
+  public void shouldUseConfiguredBackpressureAlgorithms() {
+
+    final BackpressureCfg backpressure = new BackpressureCfg();
+
+    // when
+    backpressure.setAlgorithm("gradient");
+    // then;
+    assertThat(backpressure.getAlgorithm()).isEqualTo(LimitAlgorithm.GRADIENT);
+
+    // when
+    backpressure.setAlgorithm("gradient");
+    // then;
+    assertThat(backpressure.getAlgorithm()).isEqualTo(LimitAlgorithm.GRADIENT);
+
+    // when
+    backpressure.setAlgorithm("gradient2");
+    // then;
+    assertThat(backpressure.getAlgorithm()).isEqualTo(LimitAlgorithm.GRADIENT2);
+
+    // when
+    backpressure.setAlgorithm("vegas");
+    // then;
+    assertThat(backpressure.getAlgorithm()).isEqualTo(LimitAlgorithm.VEGAS);
+
+    // when
+    backpressure.setAlgorithm("fixed");
+    // then;
+    assertThat(backpressure.getAlgorithm()).isEqualTo(LimitAlgorithm.FIXED);
+
+    // when
+    backpressure.setAlgorithm("aimd");
```

Comment:
Maybe rewrite whole test

Pseudocode
```
try (softly = new SoftAssertions()) {
  for (var algo : AlgoEnum.values()) {
   backPressure.setAlgorithm(algo.name.toLowerCase());

   softly.AssertThat(backPressure.getAlgorithm()).isEqualTo(algo);
  }
}
``` 

I know some people don't like this style. But it's less maintenance when you add new algos in the future.

Reply:
Not sure about this. What we want to test is that the strings are mapped to to the correct enum. If we use the enums to generate the string, isn't it defeating the purpose?

Reply:
sounds fair